### PR TITLE
v.db.join: handle existing columns properly

### PR DESCRIPTION
**Pull request by ninsbl**
Currently, ` v.db.join` fails if all columns to join already exist in the target table, because `v.db.addcolumn` is called with an empty list of columns to create.

This PR addresses the issue, removes Python2 related code along with some refactoring. It also adds a test-suite.

### CODE REVIEW

File: scripts/v.db.join/testsuite/test_v_db_join.py [hunk 1]
```diff
@@ -0,0 +1,76 @@
+"""
+Created on Sun Jun 09 11:28:34 2018
+
+@author: Sanjeet Bhatti
```
[the remark this answers is not in the document]
Comment (ninsbl) [reply scripts/v.db.join/testsuite/test_v_db_join.py]: THanks, good catch. I updated the header now.

The test currently only checks that the module does not fail. Better than no test, but maybe I find the time to test als results a bit...

**Comment by echoix** [comment]
For the rest, I don't really know what to look out for, and the diff (even only the first commit) is too hard to follow on a phone, I would need to look hard on a computer (but ain't available until the end of weekend). The variable renames, import alias rename, and other fixes mixed with the actual fix makes it harder to see what is equivalent or not.

**Comment by ninsbl** [comment]
Sorry. An issue got introduced earlier, with the latest update of the module I guess. If all columns to join were present in the target table, the module started failing, which was no problem earlier. The list of "columns_to_add" was empty in that case and v.db.addcolumn then fails because columns to add are missing.

**Comment by griembauer** [comment]
> Sorry. An issue got introduced earlier, with the latest update of the module I guess. If all columns to join were present in the target table, the module started failing, which was no problem earlier. The list of "columns_to_add" was empty in that case and v.db.addcolumn then fails because columns to add are missing.

Thanks a lot for catching this! The remaining refactoring also looks good to me. It ran successfully on some test tables.

**Review by griembauer — APPROVED** [review]

**Review by tmszi — APPROVED** [review]
Looks good to me. Tested and works as expected. 

Small note:

Maybe you can update module doc with that short info if columns names are same in joined table, same columns  names will be update in the source vector map table.

Alternatively, there could be a flag that would ensure the renaming of the same columns and joining them instead of updating existing columns.

**Comment by ninsbl** [comment]
> Small note:
> 
> Maybe you can update module doc with that short info if columns names are same in joined table, same columns names will be update in the source vector map table.
> 
> Alternatively, there could be a flag that would ensure the renaming of the same columns and joining them instead of updating existing columns.

Good points @tmszi will open a new issue for that and other further improvements of v.db.join.
